### PR TITLE
Lacrymology/new caspa integration

### PR DESCRIPTION
**Pull request by Lacrymology**
integrate the changes in playlists / clips models from the caspa rewrite

### CODE REVIEW

File: utils.js [hunk 1]
```diff
@@ -2,11 +2,11 @@ var moment = require('moment');
 
 exports = module.exports = {
     getXmlFileNameFromClip: function(clip) {
-        return clip.playlist_id + "-" + clip.id + ".xml";
+        return clip.playlist_id + "$" + clip.id + ".xml";
```
Comment (Lacrymology) [inline utils.js:5]: this is horrible, but it's legacy-horrible, and I don't know how to avoid it

**Comment by jmrunge** [comment]
Travis:

``` javascript
1 of 193 tests failed:

  1) PlaylistMongoDriver #subscriptions should respond to update messages:
     TypeError: Cannot read property 'pieces' of null
      at /home/travis/build/inaes-tic/mbc-mosto/drivers/playlists/mongo-driver.js:6:8093
      at /home/travis/build/inaes-tic/mbc-mosto/node_modules/mbc-common/node_modules/mongoskin/node_modules/mongodb/lib/mongodb/collection.js:957:5
      at /home/travis/build/inaes-tic/mbc-mosto/node_modules/mbc-common/node_modules/mongoskin/node_modules/mongodb/lib/mongodb/cursor.js:155:9
      at /home/travis/build/inaes-tic/mbc-mosto/node_modules/mbc-common/node_modules/mongoskin/node_modules/mongodb/lib/mongodb/cursor.js:195:11
      at /home/travis/build/inaes-tic/mbc-mosto/node_modules/mbc-common/node_modules/mongoskin/node_modules/mongodb/lib/mongodb/cursor.js:545:35
      at Cursor.close (/home/travis/build/inaes-tic/mbc-mosto/node_modules/mbc-common/node_modules/mongoskin/node_modules/mongodb/lib/mongodb/cursor.js:776:5)
      at Cursor.nextObject (/home/travis/build/inaes-tic/mbc-mosto/node_modules/mbc-common/node_modules/mongoskin/node_modules/mongodb/lib/mongodb/cursor.js:545:17)
      at commandHandler (/home/travis/build/inaes-tic/mbc-mosto/node_modules/mbc-common/node_modules/mongoskin/node_modules/mongodb/lib/mongodb/cursor.js:522:14)
      at g (events.js:175:14)
      at EventEmitter.emit (events.js:106:17)
      at Db._callHandler (/home/travis/build/inaes-tic/mbc-mosto/node_modules/mbc-common/node_modules/mongoskin/node_modules/mongodb/lib/mongodb/db.js:1472:25)
      at /home/travis/build/inaes-tic/mbc-mosto/node_modules/mbc-common/node_modules/mongoskin/node_modules/mongodb/lib/mongodb/connection/server.js:430:30
      at MongoReply.parseBody (/home/travis/build/inaes-tic/mbc-mosto/node_modules/mbc-common/node_modules/mongoskin/node_modules/mongodb/lib/mongodb/responses/mongo_reply.js:127:5)
      at null.<anonymous> (/home/travis/build/inaes-tic/mbc-mosto/node_modules/mbc-common/node_modules/mongoskin/node_modules/mongodb/lib/mongodb/connection/server.js:389:22)
      at EventEmitter.emit (events.js:95:17)
      at null.<anonymous> (/home/travis/build/inaes-tic/mbc-mosto/node_modules/mbc-common/node_modules/mongoskin/node_modules/mongodb/lib/mongodb/connection/connection_pool.js:140:13)
      at EventEmitter.emit (events.js:98:17)
      at Socket.<anonymous> (/home/travis/build/inaes-tic/mbc-mosto/node_modules/mbc-common/node_modules/mongoskin/node_modules/mongodb/lib/mongodb/connection/connection.js:321:22)
      at Socket.EventEmitter.emit (events.js:95:17)
      at Socket.<anonymous> (_stream_readable.js:736:14)
      at Socket.EventEmitter.emit (events.js:92:17)
      at emitReadable_ (_stream_readable.js:408:10)
      at emitReadable (_stream_readable.js:404:5)
      at readableAddChunk (_stream_readable.js:165:9)
      at Socket.Readable.push (_stream_readable.js:127:10)
      at TCP.onread (net.js:526:21)
```

**Comment by Lacrymology** [comment]
I understand how this failed.. what I don't understand is why it didn't fail for me locally...

**Comment by Lacrymology** [comment]
I rebased this onto the new master

**Comment by jmrunge** [comment]
I restarted build on Travis.  Still failing.  It also fails in my machine.  Same error.  Can you please take another look at this?

**Comment by Lacrymology** [comment]
I am. I've been on this for two days, and I cannot make it work properly.

**Comment by jmrunge** [comment]
I remember having this problem when reviewing @josx's PR from Backbone Relational.  I still have this branch local and this is how createPlaylist function looked:

``` javascript
mongo_driver.prototype.createPlaylist = function(sched, callback) {
    var self = this;
    var err = undefined;
    console.log("mongo-driver: [INFO] Create Playlist:", sched);

    var startDate = moment.unix(sched.start).valueOf();
    var endDate = moment.unix(sched.end).valueOf();
    var name = sched.title;
    var playlist_id = (sched._id.toHexString && sched._id.toHexString()) || sched._id;

    var medias = [];
    _.forEach(sched.list.pieces, function(piece, order) {
        var piece_id = (piece._id.toHexString && piece._id.toHexString()) || piece._id;
        var orig_order = order;
        var clip_name = piece.name;
        // TODO: don't know what goes in type
        var type = "default";
        var file = piece.file;
        var length = moment(piece.durationraw, "HH:mm:ss.SSS");
        var fps = piece.video.fps;
        medias.push(new Media(piece_id, orig_order, playlist_id, clip_name, type, file,
                              moment.duration({
                                  hours: length.hours(),
                                  minutes: length.minutes(),
                                  seconds: length.seconds(),
                                  milliseconds: length.milliseconds(),
                              }), parseFloat(fps)));
    });

    var playlist = new Playlist(playlist_id, name, startDate, medias, endDate, "snap");

    callback(err, playlist);

};
```

**Comment by Lacrymology** [comment]
For this to work, https://github.com/inaes-tic/mbc-common/pull/20 needs to be merged.

**Comment by Lacrymology** [comment]
I can't reproduce the current travis failure....

**Comment by jmrunge** [comment]
Im having the exact same error as travis does.  Deleted mbc-common and
backbone node_module's dirs and npm installed them.

2013/10/9 Tomas Neme notifications@github.com

**Comment by Lacrymology** [comment]
did you do that since yesterday? because we just merged some changes into
common

**Comment by jmrunge** [comment]
Of course I tested this after mbc-common merge.  Please take a look at my commits.  Test is passing now locally on my machine.

**Pull request merged**